src/crimson/osd/pg_backend.cc

```diff
@@ -1015,12 +1018,22 @@ PGBackend::cmp_xattr_ierrorator::future<> PGBackend::cmp_xattr(
     if (result == 0) {
       logger().info("cmp_xattr: comparison returned false");
       osd_op.rval = -ECANCELED;
+      return crimson::ct_error::ecanceled::make();
+    } else if (result == -EINVAL) {
+      return crimson::ct_error::invarg::make();
     } else {
       osd_op.rval = result;
+      return cmp_xattr_ierrorator::now();
     }
-    delta_stats.num_rd++;
-    delta_stats.num_rd_kb += shift_round_up(osd_op.op.xattr.value_len, 10);
-  });
+  }).handle_error_interruptible(
+    crimson::ct_error::enodata::handle([&delta_stats, &osd_op] ()
+      ->cmp_xattr_errorator::future<> {
+      delta_stats.num_rd++;
+      delta_stats.num_rd_kb += shift_round_up(osd_op.op.xattr.value_len, 10);
```

Comment:
Since this is unconditional, why not do it once prior to the call to getattr?

Reply:
classic osd do it after calling getattr, seems there is other error code, which will not count it.

Reply:
such as ENOENT.

Reply:
Hmm, ok.